src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -3935,13 +4011,88 @@ private void DeleteObjectInternal(object entity, bool failIfInAddedState)
                     Util.IncludeLinkState(state),
                     "bad state transition to deleted");
 
+                if (dependsOnIds != null)
+                {
+                    //In DataServiceContext.UpdateObject, we pass entity Id(s) as dependsOnIds.
+                    //We will extract ChangeOrder from the entity descriptor based on the Id.
+                    //This is because we pass the ChangeOrder as Content-ID to the batch writer.
+                    //We will pass the Content-ID(s) as dependsOnIds to the batch writer.
+                    List<string> dependsOnIdsAsChangeOrders, dependsOnChangeSetIds;
+
+                    GetDependsOnChangeOrdersAndChangeSetIds(
+                        dependsOnIds,
+                        this.EntityTracker.Entities,
+                        this.Model,
+                        out dependsOnIdsAsChangeOrders,
+                        out dependsOnChangeSetIds);
+
+                    resource.DependsOnIds = dependsOnIdsAsChangeOrders;
+                    resource.DependsOnChangeSetIds = dependsOnChangeSetIds.Distinct().ToList();

```

Comment:
Maybe a hashset would be better here. So as to avoid the Distinct checkup.

Reply:
Done